osu.Game/Rulesets/Edit/Checks/CheckMissingGenreLanguage.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
+using osu.Game.Overlays.BeatmapListing;
+using osu.Game.Rulesets.Edit.Checks.Components;
+
+namespace osu.Game.Rulesets.Edit.Checks
+{
+    public class CheckMissingGenreLanguage : ICheck
+    {
+        public CheckMetadata Metadata => new CheckMetadata(CheckCategory.Metadata, "Missing Genre/Language Tags", CheckScope.BeatmapSet);
+
+        public IEnumerable<IssueTemplate> PossibleTemplates => new IssueTemplate[]
+        {
+            new IssueTemplateMissingGenre(this),
+            new IssueTemplateMissingLanguage(this),
+        };
+
+        public IEnumerable<Issue> Run(BeatmapVerifierContext context)
+        {
+            var metadata = context.Beatmap.BeatmapInfo.Metadata;
+
+            string tags = metadata.Tags.ToLowerInvariant();
+
+            if (!hasTags<SearchGenre>(tags))
+                yield return new IssueTemplateMissingGenre(this).Create();
+
+            if (!hasTags<SearchLanguage>(tags))
+                yield return new IssueTemplateMissingLanguage(this).Create();
+        }
+
+        private bool hasTags<T>(string tags) where T : Enum
+        {
+            foreach (T value in Enum.GetValues(typeof(T)))
+            {
+                string description = getGenreLanguageString(value);
+                string[] words = description.ToLowerInvariant().Split(' ');
+
+                if (words.All(tags.Contains))
+                    return true;
+            }
+
+            return false;
+        }
+
+        // "Video Game" and "Hip Hop" are multiple words that are properly formatted in the enum's description attribute,
+        // so we need to use that and fall back to the enum's string value for the rest.
+        private string getGenreLanguageString(Enum value)
+        {
+            var field = value.GetType().GetField(value.ToString());
+            var attribute = field?.GetCustomAttribute<DescriptionAttribute>();
+            return attribute?.Description ?? value.ToString();
+        }
```

Comment:
this already exists [as an extension method](https://github.com/ppy/osu-framework/blob/a7b203bee6289e15d29490ec1d578dbc07257290/osu.Framework/Extensions/ExtensionMethods.cs#L239-L246), no need to reinvent